kernel/dev/tty/tty.cpp

```diff
@@ -188,4 +224,21 @@ void putchar(char c) {
         tty_coords_x = 0;
         tty_coords_y = X86_TTY_HEIGHT - 1;
     }
+    return c;
+}
+
+int puts(const char *str) {
+    int i = 0;
+    // Loops until a null character
+    while(str[i]) {
+        if(putchar(str[i]) == EOF) { 
```

Comment:
Maybe you just haven't finished writing the code yet, but I don't see any code in `putchar` that returns `EOF`.

Reply:
In what cases should putchar return EOF? putchar is supposed to return the character printed, so do we return EOF on null / `\n`, or in what cases?

Reply:
I have no idea, I saw that you were depending on `putchar` returning `EOF` so I was clarifying what the expected behavior should be.